shelley/chain-and-ledger/executable-spec/src/BlockChain.hs

```diff
@@ -207,54 +218,70 @@ slotsPrior = 33 -- one third of slots per epoch
 startRewards :: Duration
 startRewards = 33 -- see above
 
-verifyVrf
-  :: (DSIGNAlgorithm dsignAlgo, Eq a)
-  => VKey dsignAlgo
+-- | Construct a seed to use in the VRF computation.
+mkSeed
+  :: ( HashAlgorithm hashAlgo
+     , DSIGNAlgorithm dsignAlgo
+     , KESAlgorithm kesAlgo
+     , VRFAlgorithm vrfAlgo
+     )
+  => Nonce -- ^ Universal constant
+  -> Slot
+  -> Nonce -- ^ Epoch nonce
+  -> HashHeader hashAlgo dsignAlgo kesAlgo vrfAlgo
   -> Seed
-  -> (a, Proof dsignAlgo a)
-  -> Bool
-verifyVrf vk seed (val, Proof k s v) = vk == k && seed == s && val == v
+mkSeed (Nonce eta0) slot nonce lastHash =
```

Comment:
Maybe don't call this `eta0`, because this is the name corresponding to the epoch nonce.